include/zephyr/net/ptp.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 Carbon Robotics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
```

Comment:
I would place the changes that introduce the ptp.h to a separate commit (but part of this PR). We usually try to avoid introducing new APIs in commits that introduce a sample. So first in the PR have a commit like `net: ptp: Add support for PTP packet headers`. This so that we could revert the sample easily if needed but leave the headers etc. generic code in place.